katsdpimager/report.py

```diff
@@ -0,0 +1,356 @@
+        """
+        frequencies = frequencies.to(self.frequencies.unit, equivalencies=u.spectral())
+        ans = np.interp(frequencies, self.frequencies, self.sefd, left=np.nan, right=np.nan)
+        if effective:
+            ans /= self.correlator_efficiency
+        return ans
+
+
+class CommonStats:
+    """Information extracted from the dataset, independent of target."""
+
+    def __init__(self, dataset: katdal.DataSet, telstate: katsdptelstate.TelescopeState) -> None:
+        self.output_channels: List[int] = telstate['output_channels']
+        self.channels: int = len(dataset.channels)
+        self.channel_width = dataset.channel_width * u.Hz
+        self.frequencies: u.Quantity = dataset.freqs * u.Hz
+        self.antennas = dataset.ants
+        # XXX for now this only supports MeerKAT. MeerKAT+ will be heterogeneous.
+        self.sefd: Optional[u.Quantity] = None
+        band = dataset.spectral_windows[dataset.spw].band
+        try:
+            sefd_model = SEFDModel('meerkat', band)
+        except FileNotFoundError:
+            logger.warning('No SEFD model for band %s', band)
+        else:
+            self.sefd = sefd_model(self.frequencies, effective=True)
+
+
+class TargetStats:
+    """Collect all the statistics for a single target."""
+
+    def __init__(self, dataset: katdal.DataSet,
+                 common: CommonStats, target: katpoint.Target) -> None:
+        self.common = common
+        self.target = target
+        # Status string for each channel
+        self.status: List[str] = ['masked'] * common.channels
+        for channel in common.output_channels:
+            self.status[channel] = 'failed'
+        # Peak flux per channel (NaN where missing)
+        self.peak: u.Quantity = [math.nan] * common.channels * (u.Jy / u.beam)
+        # Noise per channel (NaN where missing)
+        self.noise: u.Quantity = [math.nan] * common.channels * (u.Jy / u.beam)
+        # Increase in noise due to imaging weights
+        self.normalized_noise = [math.nan] * common.channels
+        self.plots: Dict[str, str] = {}     # Divs to insert for plots returned by make_plots
+        self.frequency_range = bokeh.models.Range1d(
+            self.common.frequencies[0].to_value(FREQUENCY_PLOT_UNIT),
+            self.common.frequencies[-1].to_value(FREQUENCY_PLOT_UNIT),
+            bounds='auto'
+        )
+        self.channel_range = bokeh.models.Range1d(0, self.common.channels - 1, bounds='auto')
+        self.time_on_target = katsdpimager.metadata.time_on_target(dataset, target)
+        self.predicted_natural_noise: Optional[u.Quantity] = None
+        if self.common.sefd is not None and len(self.common.antennas) > 1:
+            n = len(self.common.antennas)
+            # Correlator efficiency is already folded in to self.common.sefd
+            denom = math.sqrt(2 * n * (n - 1) * self.time_on_target * self.common.channel_width)
+            self.predicted_natural_noise = self.common.sefd / denom / u.beam
```

Comment:
That bugged me a bit too. And thinking about it more now just makes my head hurt more. Maybe the mistake is just thinking about noise as a physical quantity - treat it as a statistical property and then it has the same units as the quantity it affects.

Reply:
I thought about it again, and it may also be because the noise is also convolved by the PSF: if the entire UV plane was filled with noise, then every pixel of the image would be (I think) independent noise; but the sampling in the UV plane (corresponding to the baselines) convolves both the sources and the noises, and gives the noise structure on the scale of the synthesized beam.